packages/ee/lib/service.ts

```diff
@@ -66,7 +81,51 @@ export const getIsEnterpriseEdition = async (): Promise<boolean> => {
   // if the server responds with a boolean, we return it
   // if the server errors, we return null
   // null signifies an error
-  const isValid: boolean | null = await cache(
+  const license = await fetchLicense();
+  const isValid = license ? license.status === "active" : null;
+
+  const previousResult = await getPreviousResult();
+  if (previousResult.active === null) {
+    if (isValid === null) {
+      await setPreviousResult({
+        active: false,
+        features: { isMultiOrgEnabled: false },
+        lastChecked: new Date(),
+      });
+      return false;
+    }
+  }
+
+  if (isValid !== null && license) {
+    await setPreviousResult({ active: isValid, features: license.features, lastChecked: new Date() });
+    return isValid;
+  } else {
+    // if result is undefined -> error
+    // if the last check was less than 72 hours, return the previous value:
+    if (new Date().getTime() - previousResult.lastChecked.getTime() <= 3 * 24 * 60 * 60 * 1000) {
+      return previousResult.active !== null ? previousResult.active : false;
+    }
+
+    // if the last check was more than 72 hours, return false and log the error
+    console.error("Error while checking license: The license check failed");
+    return false;
+  }
+};
+
+export const getLicenseFeatures = async (): Promise<TEnterpriseLicenseFeatures | null> => {
```

Comment:
Can we maybe write a more general `getEnterpriseLicense` function that gets the current license where we can extract the features easily. I think that would be a good and generic addition to the service.

Reply:
we have a function `fetchLicense` the returns an object containing status and features. Is that what we want here ?🤔